src/devices/ExplorerHat/Led.cs

```diff
@@ -65,21 +67,19 @@ public void Off()
 
         #region IDisposable Support
 
-        private bool _disposedValue = false;
+        private bool _shouldDispose;
 
         /// <summary>
         /// Disposes the <see cref="Led"/> instance
         /// </summary>
         protected virtual void Dispose(bool disposing)
         {
-            if (!_disposedValue)
+            if (disposing)
             {
-                if (disposing)
+                if (_shouldDispose)
```

Comment:
This is not really disposing, this is only really turning the LED off. I realize you didn't introduce this, but now that you are touching this do you mind fixing that? Just dispose the controller after calling Off. Also, I believe Off should be called regardless in this case.